common/src/main/java/tachyon/security/PlainSaslServerProvider.java

```diff
@@ -0,0 +1,86 @@
+ * the License.
+ */
+
+package tachyon.security;
+
+import java.security.Provider;
+import java.util.Map;
+
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.sasl.SaslException;
+import javax.security.sasl.SaslServer;
+import javax.security.sasl.SaslServerFactory;
+
+/**
+ * Because the Java SunSASL provider doesn't support the server-side PLAIN mechanism.
+ * There is a new provider needed to register to support server-side PLAIN mechanism.
+ * There are three basic steps in implementing a SASL security provider:
+ * 1.Write a class that implements the SaslServer interface {@link PlainSaslServer}
+ * 2.Write a factory class implements the SaslServerFactory
+ * 3.Write a JCA provider that registers the factory
+ */
+public class PlainSaslServerProvider extends Provider {
+  public static final String PROVIDER = "TachyonSaslPlain";
+  public static final String MECHANSIM = "PLAIN";
+
+  public PlainSaslServerProvider() {
+    super(PROVIDER, 1.0, "Plain SASL provider");
+    put("SaslServerFactory." + MECHANSIM, PlainSaslServerFactory.class.getName());
+  }
+
+  /**
+   * PlainSaslServerFactory was used to create instances of {@link PlainSaslServer}.
+   * The parameter mechanism must be "PLAIN" when this PlainSaslServerFactory was called, or
+   * the null will be return.
+   */
+  public static class PlainSaslServerFactory implements SaslServerFactory {
+    /**
+     * Creates a <tt>SaslServer</tt> using the parameters supplied.
+     * It returns null if no <tt>SaslServer</tt> can be created using the parameters supplied.
+     * Throws <tt>SaslException</tt> if it cannot create a <tt>SaslServer</tt>
+     * because of an error.
+     * @param mechanism The name of a SASL mechanism. (e.g. "PALIN").
+     * @param protocol The non-null string name of the protocol for which
+     * the authentication is being performed.
+     * @param serverName The non-null fully qualified host name of the server
+     * to authenticate to.
+     * @param props The possibly null set of properties used to select the SASL
+     * mechanism and to configure the authentication exchange of the selected
+     * mechanism.
+     *
+     * @param cbh The possibly null callback handler to used by the SASL
+     * mechanisms to do further operation.
+     *
+     *@return A possibly null <tt>SaslServer</tt> created using the parameters
+     * supplied. If null, this factory cannot produce a <tt>SaslServer</tt>
+     * using the parameters supplied.
+     *@exception SaslException If cannot create a <tt>SaslServer</tt> because of an error.
+     */
+    @Override
+    public SaslServer createSaslServer(String mechanism, String protocol, String serverName,
+                                       Map<String, ?> props, CallbackHandler cbh)
+                                           throws SaslException {
+      if (MECHANSIM.equals(mechanism)) {
+        return new PlainSaslServer(cbh);
```

Comment:
What is PlainSaslServer? Is that something Java already provides, or is that something you will be adding later?